tests/unit/db/design_document_tests.py

```diff
@@ -164,35 +211,251 @@ def test_delete_a_view(self):
+        ddoc.fetch()
+        self.assertIsInstance(ddoc, DesignDocument)
+        self.assertEqual(ddoc, data)
+        self.assertIsInstance(ddoc['indexes']['index001'], dict)
+
+    def test_fetch_text_indexes_and_query_views(self):
+        """
+        Ensure that the document fetch from the database returns the
+        DesignDocument format as expected when retrieving a design document
+        containing query index views and text index definitions.
+        """
+        # This is not the preferred way of dealing with query index
+        # views but it works best for this test.
+        data = {
+            '_id': '_design/ddoc001',
+            'language': 'query',
+            'views': {
+                'view001': {'map': {'fields': {'name': 'asc', 'age': 'asc'}},
+                            'reduce': '_count',
+                            'options': {'def': {'fields': ['name', 'age']},
+                                        'w': 2}
+                            }
+                    },
+            'indexes': {'index001': {
+                'index': {'index_array_lengths': True,
+                          'fields': [{'name': 'name', 'type': 'string'},
+                                     {'name': 'age', 'type': 'number'}],
+                          'default_field': {'enabled': True,
+                                            'analyzer': 'german'},
+                          'default_analyzer': 'keyword',
+                          'selector': {}},
+                'analyzer': {'name': 'perfield',
+                             'default': 'keyword',
+                             'fields': {'$default': 'german'}}}}}
+        doc = self.db.create_document(data)
+        self.assertIsInstance(doc, Document)
+        data['_rev'] = doc['_rev']
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        ddoc.fetch()
+        self.assertIsInstance(ddoc, DesignDocument)
+        self.assertEqual(ddoc, data)
+        self.assertIsInstance(ddoc['indexes']['index001'], dict)
+        self.assertIsInstance(ddoc['views']['view001'], QueryIndexView)
+
+    def test_save_fails_query_view_validation(self):
+        """
+        Ensure that all views are query views when language is set to query.
+        """
+        ddoc = DesignDocument(self.db, '_design/ddoc001')
+        view_map = 'function (doc) {\n  emit(doc._id, 1);\n}'
+        view_reduce = '_count'
+        db_copy = '{0}-copy'.format(self.db.database_name)
+        ddoc.add_view('view001', view_map, view_reduce, dbcopy=db_copy)
+        ddoc['language'] = 'query'
+        with self.assertRaises(CloudantException) as cm:
+            ddoc.save()
+        err = cm.exception
+        self.assertEqual(
+            str(err),
+            'View view001 must be of type QueryIndexView.'
+        )
+
+    def test_save_passes_query_view_validation(self):
+        """
+        Ensure that all views are query views when language is set to query.
```

Comment:
I don't understand this comment because it looks to me like this test is saving a MR view and the language is not set to query (unless it is by default)?